### PR TITLE
BAEL-6355 - OpenAI API with Spring Boot

### PR DESCRIPTION
**Pull request by abh1navv**

### CODE REVIEW

File: spring-boot-modules/spring-boot-libraries-2/src/main/java/com/baeldung/openapi/controller/ChatController.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                apiUrl,
+                request,
+                ChatResponse.class);
+
+        if (response == null || response.getChoices() == null || response.getChoices().isEmpty()) {
+            return "No response";
+        }
+
+        // return the first response
+        return response.getChoices().get(0).getMessage().getContent();
```
Comment (Alexenon) [inline spring-boot-modules/spring-boot-libraries-2/src/main/java/com/baeldung/openapi/controller/ChatController.java:42]: Will be great to remove single comments and replace them with javadoc

File: spring-boot-modules/spring-boot-libraries-2/src/main/java/com/baeldung/openapi/dto/Message.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    private String role;
+    private String content;
+
+    Message(String role, String content) {
+        this.role = role;
+        this.content = content;
+    }
+
+    Message() {
```
Comment (Alexenon) [inline spring-boot-modules/spring-boot-libraries-2/src/main/java/com/baeldung/openapi/dto/Message.java:13]: Do you need an empty constructor ?

**Review by Alexenon — APPROVED** [review]

File: spring-boot-modules/spring-boot-libraries-2/src/main/java/com/baeldung/openapi/controller/ChatController.java (hunk 1 above)
Comment (abh1navv) [reply spring-boot-modules/spring-boot-libraries-2/src/main/java/com/baeldung/openapi/controller/ChatController.java:42]: Updated

File: spring-boot-modules/spring-boot-libraries-2/src/main/java/com/baeldung/openapi/dto/Message.java (hunk 2 above)
Comment (abh1navv) [reply spring-boot-modules/spring-boot-libraries-2/src/main/java/com/baeldung/openapi/dto/Message.java:13]: it was required for deserializing the response

**Pull request merged**